be/src/exprs/runtime_filter.cpp

```diff
@@ -97,14 +99,19 @@ size_t SimdBlockFilter::deserialize(const uint8_t* data) {
 #undef SIMD_BF_COPY_FIELD
     const size_t alloc_size = get_alloc_size();
     DCHECK(data_size == alloc_size);
-    const int malloc_failed = posix_memalign(reinterpret_cast<void**>(&(_directory)), 64, alloc_size);
-    if (malloc_failed) throw ::std::bad_alloc();
-    memcpy(_directory, data + offset, data_size);
-    offset += data_size;
+    if (LIKELY(data_size > 0)) {
```

Comment:
What will happen, when the sender BE with this new version and the receiver with the old version ,which doesn't check whether `data_size` is positive and always creates a `_directory`?

Reply:
Should we define a new `TExecPlanFragmentParams::func_version` and only apply the optimization of this PR when FE sets the plan with the new version?

Reply:
Maybe there isn't problem. The `TqueryOptions::global_runtime_filter_build_max_size` could achieve the same goal of a new TExecPlanFragmentParams::func_version.

Reply:
the old FE won't pass `global_runtime_filter_build_max_size` and `global_runtime_filter_build_max_size` used by BE will be INT64_MAX by default. 
It means that this restriction won't take effect during upgrading, so data_size will always be greater than 0, which is safe
```
    size_t global_runtime_filter_build_max_size = UINT64_MAX;
    if (query_options.__isset.global_runtime_filter_build_max_size &&
        query_options.global_runtime_filter_build_max_size > 0) {
        global_runtime_filter_build_max_size = query_options.global_runtime_filter_build_max_size;
    }

```